thermo_nerf/thermal_nerf/thermal_nerf_model.py

```diff
@@ -348,7 +349,10 @@ def get_image_metrics_and_images(
         )
         combined_thermal = torch.cat([gt_thermal, thermal], dim=1)
 
-        images_dict.update({"thermal": combined_thermal})
+        images_dict.update({RenderedImageModality.THERMAL.__str__(): thermal})
```

Comment:
Don't use `__str__` like this directly. Instead in your case, the simpler way is to do

```python
images_dict.update({RenderedImageModality.THERMAL.value: thermal})
images_dict.update(
            {RenderedImageModality.THERMAL_COMBINED.value: combined_thermal}
        )
```

But do you need to use a string?


```python
images_dict.update({RenderedImageModality.THERMAL: thermal})
images_dict.update(
            {RenderedImageModality.THERMAL_COMBINED: combined_thermal}
        )
```
 Should work right?

Reply:
I get this `Argument of type "dict[RenderedImageModality, Tensor]" cannot be assigned to parameter "m" of type "Iterable[tuple[str, Tensor]]" in function "update"` from pylance and therefore I think we need a string here.

Reply:
It's because the dict typing annotation says the key is a string and now you are forcing an enum... 

Pylance here is not correct ;). But it's ok let's keep a string